sycl/source/detail/ur.cpp

```diff
@@ -300,7 +300,8 @@ static bool checkELFSectionPresent(const std::string &ExpectedSectionName,
 
   // End early if we do not have the expected number of section headers or
   // if the read section string header index is out-of-range.
-  if (ImgSize < SectionHeaderOffset + SectionHeaderNum * SectionHeaderSize ||
+  if (ImgSize < SectionHeaderOffset +
+                    (uint64_t)SectionHeaderNum * SectionHeaderSize ||
```

Comment:
can we make this a static_cast instead?

Reply:
I've gone ahead and used static cast instead, although may I ask why? I am assuming it is because static_cast is safer.

Reply:
yeah it's safer and it's also in the LLVM coding standards: [prefer c++ style casts](https://llvm.org/docs/CodingStandards.html#prefer-c-style-casts)